### PR TITLE
feat: create migration for old discount

### LINKED ISSUE #6690: Create migration for old discount codes to new promotions-discount
This will allow us to completely remove the old discount code code. This work can't be done until we do Milestone 2 (Coupons) though

### PR DESCRIPTION
**Pull request by vanpho93**
Resolves #6690
Impact: **major**
Type: **feature**

## Issue

Old discount codes will not work in new promotion framework

## Solution

Create migration for carts / discount codes to new promotions / coupons

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
   },
   "engineStrict": true,
   "scripts": {
-    "start:dev": "npm run start:dev -w apps/reaction",
+    "start:dev": "pnpm --filter=reaction run start:dev",
```
Comment (brent-hoover) [inline package.json]: Generally we should not just throw in extra changes unrelated to solving a particular problem. We should try to keep PR's as focused as possible. Not blocking on this one though.

File: packages/api-plugin-promotions-coupons/migrations/2.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import getCurrentShopTime from "./getCurrentShopTime.js";
+
+/**
+ * @summary returns an auto-incrementing integer id for a specific entity
+ * @param {Object} db - The db instance
+ * @param {String} shopId - The shop ID
+ * @param {String} entity - The entity (normally a collection) that you are tracking the ID for
+ * @return {Promise<Number>} - The auto-incrementing ID to use
+ */
+async function incrementSequence(db, shopId, entity) {
```
Comment (brent-hoover) [inline packages/api-plugin-promotions-coupons/migrations/2.js:12]: We have a function defined in the code already for this, we should not duplicate it here.

File: packages/api-plugin-promotions-coupons/migrations/2.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+/**
+ * @summary Performs migration down from previous data version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function down({ progress }) {
```
Comment (brent-hoover) [inline packages/api-plugin-promotions-coupons/migrations/2.js]: For this to be a proper down, it should reverse the above migration

File: packages/api-plugin-promotions-coupons/migrations/migrationsNamespace.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export const migrationsNamespace = "promotion-coupons";
```
Comment (brent-hoover) [inline packages/api-plugin-promotions-coupons/migrations/migrationsNamespace.js:1]: So I am not seeing the check for version but we need to do that, but ONLY when they have existing discounts created.

**Review by brent-hoover — CHANGES_REQUESTED** [review]

File: packages/api-plugin-promotions-coupons/migrations/2.js (hunk 2 above)
Comment (vanpho93) [reply packages/api-plugin-promotions-coupons/migrations/2.js:12]: The defined function is in another plugin, and we only have access to ({ db, progress }) so it's not easy to do.

**Review by brent-hoover — APPROVED** [review]

**Pull request merged**